modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/servlet/taglib/InstanceSettingsClientExtensionsDynamicInclude.java

```diff
@@ -0,0 +1,244 @@
+
+			InstanceSettingsClientExtensionsConfiguration
+				instanceSettingsClientExtensionsConfiguration =
+					_getInstanceSettingsClientExtensionsConfiguration();
+
+			String[] globalJSClientExtensionsExternalReferenceCodes =
+				instanceSettingsClientExtensionsConfiguration.
+					getJSClientExtensionsExternalReferenceCodes();
+
+			List<GlobalJSCET> globalJSCETs = _filterByScriptLocation(
+				_fetchGlobalJSCETs(
+					globalJSClientExtensionsExternalReferenceCodes),
+				scriptLocation);
+
+			for (GlobalJSCET globalJSCET : globalJSCETs) {
+				Map<String, String> scriptAttributes = _toAttributesMap(
+					globalJSCET, httpServletRequest);
+
+				String htmlJSScript = HTMLUtil.toHTMLJSScript(
+					scriptAttributes,
+					_toScriptElementAttributes(
+						globalJSCET.getScriptElementAttributesJSON()));
+
+				printWriter.print(htmlJSScript);
+			}
+		}
+		catch (ConfigurationException configurationException) {
+			_log.error(configurationException);
+
+			throw new RuntimeException(configurationException);
+		}
+	}
+
+	private List<GlobalJSCET> _fetchGlobalJSCETs(
+		String[] globalJSClientExtensionsExternalReferenceCodes) {
+
+		List<GlobalJSCET> globalJSCETS = new ArrayList<>();
+
+		for (String externalRefenceCode :
+				globalJSClientExtensionsExternalReferenceCodes) {
+
+			GlobalJSCET globalJSCET = _getGlobalJSCET(externalRefenceCode);
+
+			if (Objects.nonNull(globalJSCET)) {
+				globalJSCETS.add(globalJSCET);
+			}
+		}
+
+		return globalJSCETS;
+	}
+
+	private List<GlobalJSCET> _filterByScriptLocation(
+		List<GlobalJSCET> globalJSCETS, String scriptLocation) {
+
+		List<GlobalJSCET> filteredGlobalJSCETs = new ArrayList<>();
+
+		for (GlobalJSCET globalJSCET : globalJSCETS) {
+			if (Objects.equals(
+					_getCETScriptLocation(globalJSCET), scriptLocation)) {
+
+				filteredGlobalJSCETs.add(globalJSCET);
+			}
+		}
+
+		return filteredGlobalJSCETs;
+	}
+
+	private String _getCETScriptLocation(GlobalJSCET cet) {
+		return UnicodePropertiesBuilder.fastLoad(
+			cet.getTypeSettings()
+		).build(
+		).get(
+			"scriptLocation"
+		);
+	}
+
+	private GlobalJSCET _getGlobalJSCET(String externalReferenceCode) {
+		return (GlobalJSCET)_cetManager.getCET(
+			CompanyThreadLocal.getCompanyId(), externalReferenceCode);
+	}
+
+	private InstanceSettingsClientExtensionsConfiguration
+			_getInstanceSettingsClientExtensionsConfiguration()
+		throws ConfigurationException {
+
+		return _configurationProvider.getCompanyConfiguration(
+			InstanceSettingsClientExtensionsConfiguration.class,
+			CompanyThreadLocal.getCompanyId());
+	}
+
+	private Map<String, String> _toAttributesMap(
+		GlobalJSCET globalJSCET, HttpServletRequest httpServletRequest) {
+
+		UnicodeProperties typeSettingsUnicodeProperties =
+			UnicodePropertiesBuilder.fastLoad(
+				globalJSCET.getTypeSettings()
+			).build();
+
+		return HashMapBuilder.put(
+			"loadType",
+			() -> {
+				String loadType = typeSettingsUnicodeProperties.getProperty(
```

Comment:
We don't have "load type" in typeSettings for the globalJSCET. That feature is only for `ClientExtensionEntryRel`. We have to get that info from the configuration. The idea is to have the configuration stored like this:
```
[
    "ERC_[default | async | defer]_[head | bottom]"
]
```

ERC = External Reference Code

Some examples:

```
"12345-abc45-82jasdfjx82_async_head",
"12345-abc45-82jasdfjx82_default_bottom"
"12345-abc45-82jasdfjx82_defer_bottom"
```

Reply:
@thiago-buarqque do we follow a pattern like this in other places in portal?

Reply:
Yep, my suggestion is based on the current behavior when selection cxs for a layout
https://github.com/liferay/liferay-portal/blob/14288b1f8b2c29bbadcc79b3086a1aefd0f2269a/modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/layout/look_and_feel/GlobalJSCETsConfiguration.tsx#L227

The value is sent to the backend and then read to create a `ClientExtensionEntryRel` the load type and script location is added to the type settings of that entity

Reply:
Hey @thiago-buarqque Well in that case it's only formatted like that by the frontend. Since the backend splits it out and persists it as typeSettings, shouldn't we also think about persisting this in a more formal way? Am I understanding correctly what you're saying... to me it sounds like you're saying we would put "12345-abc45-82jasdfjx82_async_head" in the database? Is that what you meant?

Reply:
Yep, that's what I meant. I just had a thought regarding how would the dev import that config via a configuration cx. Because of that, do you think it would be better to translate that to a JSON string?

```
[
    {"ERC": "...", "location": "head", "loadType": "async"},
    // ...
]

```

Reply:
Other than that I'd have to research on how to have an entity to map that values (an entity similar to `ClientExtensionEntryRel`, if that's even possible for the configuration scenario

Reply:
JSON seems reasonable to me. That would be easier to create on the frontend as well when we are saving the configuration.

Reply:
Also, we need to include the value of `restrictedToAdmin` in this as well. https://www.figma.com/design/R3aOQbltTsDbfDlGiWC2Sl/lpd-33748-new-js-client-extension-instance-level?node-id=35-5370&t=J6phZpvptNwstwl9-0

Reply:
The issue with the JSON string approach is how that would be when using a configuration client extension to set that value. The idea of asking the dev to create JSON string to add that config doesn't make me happy. I am currently checking if it's possible to not make that easier

Reply:
My guess is that we could have a way to translate the YAML into JSON and then store it